packages/core/upload/admin/src/index.js

```diff
@@ -27,15 +27,27 @@ export default {
         defaultMessage: 'Media Library',
       },
       permissions: PERMISSIONS.main,
-      async Component() {
-        const component = await import(/* webpackChunkName: "upload" */ './pages/App');
-
-        return component;
-      },
+      Component: () => import(/* webpackChunkName: "upload" */ './pages/App'),
     });
 
-    app.addFields({ type: 'media', Component: MediaLibraryInput });
-    app.addComponents([{ name: 'media-library', Component: MediaLibraryDialog }]);
+    app.addFields({
+      type: 'media',
+      Component: React.lazy(() =>
+        import('./components/MediaLibraryInput').then((module) => ({
+          default: module.MediaLibraryInput,
+        }))
+      ),
```

Comment:
How come these ones still need to be `React.lazy` can we not unify the API?

Reply:
I agree. I'll revert the change for now, because I'd like to get this PR on the road and changing these (yet undocumented) APIs will require a more and different testing than I have done so far.